xgcm/grid.py

```diff
@@ -75,6 +87,21 @@ def __init__(self, ds, axis_name, periodic=True, default_shifts={}, coords=None)
         coords : dict, optional
             Mapping of axis positions to coordinate names
             (e.g. `{'center': 'XC', 'left: 'XG'}`)
+        boundary : str or dict, optional,
+            boundary can either be one of  {None, 'fill', 'extend', 'extrapolate'}
+            the above strings (in which case it is set as boundary for all axes)
+            or a dict mapping axis name to a string (e.g. {"X": "fill", "Y": "extend"}).
```

Comment:
Can we note here that these are just defaults, and that boundary can still be overwritten when calling specific methods.

Reply:
Also, does it make sense to allow a dict at the Axis level? Does each Axis have only one possible axis name?

Reply:
no this was a bad copy paste job. there is an error check is boundary is not None or one of the valid strings.